app/models/mixins/supports_feature_mixin.rb

```diff
@@ -265,15 +168,17 @@ def unsupported
 
     # use this for making a class not support a feature
     def unsupported_reason_add(feature, reason = nil)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
+      guard_queryable_feature(feature)
       feature = feature.to_sym
       unsupported[feature] = SupportsFeatureMixin.reason_or_default(reason)
     end
 
-    def define_supports_feature_methods(feature, is_supported: true, reason: nil, &block)
+    def define_supports_feature_methods(feature, is_supported: true, description: nil, reason: nil, &block)
       method_name = "supports_#{feature}?"
       feature = feature.to_sym
 
+      queryable_features[feature] = description
```

Comment:
So, now that everything is refactored in this PR, it seems to me that the `description` being stored is *never* actually used anywhere in this code.  So, now I strongly believe that we should remove `description` altogether.

Going further, the only places where `queryable_features` are used are in the `feature_known?` and `guard_queryable_feature` methods.  The only one meant to be public is `feature_known?`, but I question the value of this method in the interface.  A quick search reveals that there are [no callers](https://github.com/search?q=org%3AManageIQ+%22feature_known%3F%22&type=Code) of this method.  So, advocating that we get rid of it.

Reply:
Ok - working on removing `description`, `queryable_features` and `feature_known?`.....

Reply:
Done!